### PR TITLE
Add Function to Predict License Minting Fee in Licensing Module

### LINKED ISSUE #240: Add Function to Predict Minting License Fee
## Description

We need to add a new function to the `LicensingModule` that allows users to predict the minting fee for a given license and IP. This enhancement will provide better transparency and planning for users by allowing them to estimate the costs associated with minting license tokens.

## Rationale

Currently, users do not have a straightforward way to estimate the minting fees before actually minting the license tokens. By adding a function to predict these fees, users can make more informed decisions and plan their actions accordingly.

## Acceptance Criteria

- The `predictMintingLicenseFee` function is added to the `LicensingModule` contract.
- The function correctly calculates and returns the predicted minting fee based on the provided parameters.
- Unit tests are added to verify the correctness of the function.

### PR DESCRIPTION
**Pull request by kingster-will**
## Description

This PR introduces a new function to the Licensing Module that allows users to predict the minting fee for a given license and IP. This enhancement provides better transparency and planning for users by allowing them to estimate the costs associated with minting license tokens.

## Key Changes

- **New Function**: Added a function to predict the minting fee for a given license and IP.

This enhancement is part of our ongoing efforts to improve the functionality and user experience of the Licensing Module.

## Related Issues:
Closes #240

### CODE REVIEW

**Comment by Ramarti** [comment]
Is this issue raised in the audit related?
https://github.com/storyprotocol/protocol-core-v1/issues/183

**Comment by kingster-will** [comment]
> Is this issue raised in the audit related? #183
This PR is mainly for a new requirement that allow App can show amount of minting fee the user would pay before user decides to mint the license token.

We will have another PR for addressing the #183

But we can consider the PR for address requirement in the similar direction of issue #183

File: contracts/interfaces/modules/licensing/ILicensingModule.sol [hunk 1]
```diff
@@ -138,4 +138,21 @@ interface ILicensingModule is IModule {
...
     ) external;
+
+    /// @notice pre-compute the minting license fee for the given IP and license terms.
+    /// the function can be used to calculate the minting license fee before minting license tokens.
+    /// @param licensorIpId The IP ID of the licensor.
+    /// @param licenseTemplate The address of the license template.
+    /// @param licenseTermsId The ID of the license terms.
+    /// @param amount The amount of license tokens to mint.
+    /// @param receiver The address of the receiver.
+    /// @param royaltyContext The context of the royalty.
```
Comment (sebsadface) [inline contracts/interfaces/modules/licensing/ILicensingModule.sol:149]: Please add return values in the comments here for clarity.

File: contracts/modules/licensing/LicensingModule.sol [hunk 2]
```diff
@@ -382,6 +382,61 @@ contract LicensingModule is
...
     }
 
+    /// @notice pre-compute the minting license fee for the given IP and license terms.
+    /// the function can be used to calculate the minting license fee before minting license tokens.
+    /// @param licensorIpId The IP ID of the licensor.
+    /// @param licenseTemplate The address of the license template.
+    /// @param licenseTermsId The ID of the license terms.
+    /// @param amount The amount of license tokens to mint.
+    /// @param receiver The address of the receiver.
+    /// @param royaltyContext The context of the royalty.
```
Comment (sebsadface) [inline contracts/modules/licensing/LicensingModule.sol:392]: Same here, please add return values in the comments.

File: contracts/interfaces/modules/licensing/ILicensingModule.sol (hunk 1 above)
Comment (kingster-will) [reply contracts/interfaces/modules/licensing/ILicensingModule.sol:149]: Thanks for review! 
Added comments.

File: contracts/modules/licensing/LicensingModule.sol (hunk 2 above)
Comment (kingster-will) [reply contracts/modules/licensing/LicensingModule.sol:392]: Added comments.

**Review by sebsadface — APPROVED** [review]
LGTM :)

**Pull request merged**